### PR TITLE
ydk 0.5.2 (new formula)

### PR DESCRIPTION
**Pull request by ylil93**
alpha release

- [x ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/ydk.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      mkdir("build")
+      cd "build" do
+        system "cmake", "..", *std_cmake_args
+        system "make", "install"
+      end
+    end
+  end
+
+  test do
+    system "brew", "ls", "--versions", "ydk"
```
Comment (woodruffw) [inline Formula/ydk.rb]: We need a better test than this, something that invokes `ydk` itself in a non-trivial way (i.e., no `--help` or `--version`).

**Comment by woodruffw** [comment]
Possibly a PCRE dependency?

```
CMake Error at /usr/local/Cellar/cmake/3.7.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:138 (message):
  Could NOT find PCRE (missing: PCRE_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.7.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  CMakeModules/FindPCRE.cmake:25 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:132 (find_package)
```

Lots of boost warnings as well, although they don't appear to be fatal for the time being...

**Review by woodruffw — CHANGES_REQUESTED** [review]

File: Formula/ydk.rb (hunk 1 above)
Comment (ylil93) [inline Formula/ydk.rb]: Thanks for the feed back.
Updated the dependencies and test sections.

**Comment by ylil93** [comment]
I noticed that after updating PCRE as a dependency, the Jenkins build is still failing - any suggestions on this?

**Comment by woodruffw** [comment]
Just some minor `audit` problems now:

```
Error: 6 problems in 1 formula
ydk:
  * Use "#{ENV.cxx}" instead of hard-coding "g++"
  * C: 35: col 96: Trailing whitespace detected.
  * C: 36: col 91: Trailing whitespace detected.
  * C: 37: col 92: Trailing whitespace detected.
  * C: 38: col 95: Trailing whitespace detected.
  * C: 39: col 92: Trailing whitespace detected.
```

**Comment by ylil93** [comment]
changes made 👌

**Comment by abhikeshav** [comment]
@woodruffw The requested changes have been made and looks like the CI has passed as well. Can this be merged?

Thanks!

File: Formula/ydk.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+class Ydk < Formula
+  desc "generate API bindings to YANG data models"
+  homepage "https://github.com/abhikeshav/ydk-cpp/blob/master/README.md"
```
Comment (woodruffw) [inline Formula/ydk.rb]: Just `https://github.com/abhikeshav/ydk-cpp`, please.

File: Formula/ydk.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  test do
+    (testpath/"test.cpp").write <<-EOS.undent
+      #include <ydk/path_api.hpp>
+      int main() {
+        ydk::path::Repository repo{};
+        return 0;
+      }
+    EOS
+    system ENV.cxx, "-std=c++11", "-Wall", "-Wextra", "-g", "-O0", "test.cpp", "-otest", "-lboost_log_setup-mt", "-lboost_log-mt", "-lboost_thread-mt", "-lboost_date_time-mt", "-lboost_system-mt", "-lboost_filesystem-mt", "-lboost_log_setup-mt", "-lboost_log-mt", "-lboost_thread-mt", "-lboost_date_time-mt", "-lboost_system-mt", "-lboost_filesystem-mt", "-lboost_log_setup-mt", "-lboost_log-mt", "-lboost_filesystem-mt", "-lboost_system-mt", "-lxml2", "-lcurl", "-lssh_threads", "-lpcre", "-lxslt", "-lssh", "-lpthread", "-ldl", "-lydk"
```
Comment (woodruffw) [inline Formula/ydk.rb]: Break this into multiple lines.

**Comment by woodruffw** [comment]
@abhikeshav I'm a little bit confused by `ydk-cpp`'s ownership. Is packaging this fork intentional, or did @ylil93 intend to package @CiscoDevNet's?

We'll likely require the parent fork here if they control the name (and if that's what people expect).

**Comment by abhikeshav** [comment]
Thanks for your reply. Yes. @Ciscodevnet is the intended owner

**Comment by woodruffw** [comment]
The `url` needs to be updated to their copy of the repo, then.

**Comment by ylil93** [comment]
changes made

File: Formula/ydk.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  depends_on "cmake" => :build
+  depends_on "boost"
+  depends_on "boost-python"
+  depends_on "curl"
+  depends_on "libssh"
+  depends_on "pcre"
+  depends_on "xml2"
+  depends_on "pkg-config" => :build
+  depends_on :x11 => :optional
```
Comment (MikeMcQuaid) [inline Formula/ydk.rb]: Let's remove this, we're trying to reduce options and particularly X11 ones, thanks.

File: Formula/ydk.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  def install
+    cd "core/ydk" do
```
Comment (MikeMcQuaid) [inline Formula/ydk.rb]: `mkdir "code/ydk/build" do` should do this more simply

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by ylil93** [comment]
Thanks for taking the time to review our formula, but we've decided to host our own tap instead, so you can close this request now.

Thanks again for all the input.

**Pull request closed without merging**